### PR TITLE
Final Grade section in staff tools

### PR DESCRIPTION
**Pull request by cahrens**
## [TNL-3761](https://openedx.atlassian.net/browse/TNL-3761)

Adds more details to the "Final Grade" section in instructor tools. See screenshots below for more details.

Also fixes TNL-3930.

### Sandbox
- [x] https://finalgrade.sandbox.edx.org/

### Testing
- [ ] i18n - need to push/pull strings, as I've changed some display values
- [ ] RTL - will verify with i18n
- [x] Unit and acceptance tests added
- [x] Accessibility tests passing

### Reviewers
If you've been tagged for review, please check your corresponding box once you've given the :+1:.
- [x] Code review: @efischer19  
- [ ] Code review: @dianakhuang 
- [ ] UI strings/error msgs review: @catong 

### Post-review
- [ ] Squash commits

### CODE REVIEW

**Comment by cahrens** [comment]
Screenshots showing different states:

1. Staff grade, peer grade column with 2 peer responses

image

File: openassessment/templates/openassessmentblock/staff_area/oa_student_info.html [hunk 1]
```diff
@@ -240,21 +240,61 @@ <h2 class="staff-info__title ui-staff__subtitle ui-toggle-visibility__control">
...
                     {% with points_earned_string=score.points_earned|stringformat:"s" points_possible_string=score.points_possible|stringformat:"s" %}
                         {% blocktrans with points_earned='<span class="grade__value__earned">'|safe|add:points_earned_string|add:'</span>'|safe points_possible='<span class="grade__value__potential">'|safe|add:points_possible_string|add:'</span>'|safe %}
                             Final grade: {{ points_earned }} out of {{ points_possible }}
                         {% endblocktrans %}
                     {% endwith %}
                     </p>
+                    <table class="staff-info__status__table staff-info__final__grade__table" summary="{% trans "Final Grade Details" %}">
+                        <thead>
+                        <tr>
+                            <th abbr="Criterion" scope="col">{% trans "Criterion" %}</th>
```
Comment (cahrens) [inline openassessment/templates/openassessmentblock/staff_area/oa_student_info.html]: I should translate the "abbr" value here.

File: openassessment/xblock/test/test_staff_area.py [hunk 2]
```diff
@@ -233,6 +235,9 @@ def test_staff_area_student_info_self_only(self, xblock):
         self.assertEquals("Bob Answer 1", context['submission']['answer']['parts'][0]['text'])
         self.assertEquals(None, context['peer_assessments'])
         self.assertEquals("openassessmentblock/staff_area/oa_student_info.html", path)
+        grade_details = context['grade_details']
```
Comment (cahrens) [inline openassessment/xblock/test/test_staff_area.py]: Note that I didn't attempt to cover every permutation in unit tests. I am reusing a helper method, and the acceptance test coverage is pretty strong.

File: openassessment/templates/openassessmentblock/staff_area/oa_student_info.html [hunk 3]
```diff
@@ -240,21 +240,61 @@ <h2 class="staff-info__title ui-staff__subtitle ui-toggle-visibility__control">
...
+                            <tr>
+                                <td class="label">{{ criterion.label }}</td>
+                                {% for assessment in criterion.assessments %}
+                                    <td class="value">
+                                    {% if assessment.points != None %}
+                                        <div>
+                                        {% blocktrans with assessment_label=assessment.option.label count points=assessment.points %}
+                                            {{ assessment_label }} - {{ points }} point
+                                        {% plural %}
+                                            {{ assessment_label }} - {{ points }} points
```
Comment (efischer19) [inline openassessment/templates/openassessmentblock/staff_area/oa_student_info.html]: Will this get RTL-ed properly? Let me check it out on a sandbox quickly...

File: openassessment/xblock/staff_area_mixin.py [hunk 4]
```diff
@@ -378,18 +378,49 @@ def get_student_info_path_and_context(self, student_username, expanded_view=None
         self_assessment = None
         peer_assessments = None
         submitted_assessments = None
+        peer_assessments_grade_context = []
+        self_assessment_grade_context = None
+        example_based_assessment_grade_context = None
+        staff_assessment_grade_context = None
```
Comment (efischer19) [inline openassessment/xblock/staff_area_mixin.py]: Would it be worthwhile to pair all these `XXX_assessment` and `XXX_assessment_context` item together in tuples?

File: openassessment/xblock/staff_area_mixin.py [hunk 5]
```diff
@@ -378,18 +378,49 @@ def get_student_info_path_and_context(self, student_username, expanded_view=None
...
 
         if "example-based-assessment" in assessment_steps:
             example_based_assessment = ai_api.get_latest_assessment(submission_uuid)
-
-        workflow = self.get_workflow_info(submission_uuid=submission_uuid)
+            if grade_exists:
+                example_based_assessment_grade_context = self._assessment_grade_context(example_based_assessment)
+
+        if grade_exists:
+            staff_assessment = staff_api.get_latest_staff_assessment(submission_uuid)
```
Comment (efischer19) [inline openassessment/xblock/staff_area_mixin.py]: I don't think `staff_assessment` is defined above, we'll need to coordinate this with my in-flight PR once one of them merges.

**Comment by efischer19** [comment]
Provisional :+1: from me, I want to verify the RTL-correctness of the string I noted above, and we'll need some coordination between our two PRs once one of them lands.

File: openassessment/xblock/staff_area_mixin.py (hunk 4 above)
Comment (cahrens) [inline openassessment/xblock/staff_area_mixin.py]: I grouped them together now, though they are not actually tuples.

File: openassessment/xblock/staff_area_mixin.py (hunk 5 above)
Comment (cahrens) [inline openassessment/xblock/staff_area_mixin.py]: I've rebased (haven't pushed up yet).

File: openassessment/templates/openassessmentblock/staff_area/oa_student_info.html (hunk 3 above)
Comment (cahrens) [inline openassessment/templates/openassessmentblock/staff_area/oa_student_info.html]: Talked to Eric, we will do more testing after we push the translations.

**Comment by cahrens** [comment]
@andy-armstrong and @catong Can you review this PR today? It has been rebased to pick up Eric's changes, and I just updated my sandbox and verified that acceptance tests are passing.

**Comment by cahrens** [comment]
@dianakhuang Could you review this instead of @andy-armstrong?

File: openassessment/xblock/staff_area_mixin.py [hunk 6]
```diff
@@ -375,22 +375,56 @@ def get_student_info_path_and_context(self, student_username, expanded_view=None
...
+
         staff_assessment = staff_api.get_latest_staff_assessment(submission_uuid)
+        staff_assessment_grade_context = None
+
+        submitted_assessments = None
+
+        grade_details = None
+
+        workflow = self.get_workflow_info(submission_uuid=submission_uuid)
+        grade_exists = workflow.get('status') == "done"
```
Comment (dianakhuang) [inline openassessment/xblock/staff_area_mixin.py]: Do we not have an constant somewhere that defines this `"done"`?

**Comment by dianakhuang** [comment]
Other than that one minor comment, :+1:

File: openassessment/xblock/staff_area_mixin.py (hunk 6 above)
Comment (cahrens) [inline openassessment/xblock/staff_area_mixin.py]: It doesn't look like it-- I see a lot of places in the code where the workflow statuses are compared/iterated through, and not any constants. I'm not feeling particularly inclined introduce constants at this time.

Comment (dianakhuang) [inline openassessment/xblock/staff_area_mixin.py]: That sounds reasonable to me.

**Comment by catong** [comment]
@cahrens :+1: from me.

File: openassessment/templates/openassessmentblock/staff_area/oa_student_info.html (hunk 3 above)
Comment (efischer19) [inline openassessment/templates/openassessmentblock/staff_area/oa_student_info.html]: Screencap of my sandbox with 0.2.6 translations being pushed:

screen shot 2016-01-21 at 4 23 12 pm

Is this okay?I think translators will be able to translate the `blocktrans` above into RTL languages properly, so I'm good with this if everyone else is.

Comment (cahrens) [inline openassessment/templates/openassessmentblock/staff_area/oa_student_info.html]: I think it is OK. We may learn something relevant when we fix the RTL bug you found, and if so, we can update this similar usages then.

**Pull request merged**